packages/spotlight/SpotlightContainerDecorator/SpotlightContainer.js

```diff
@@ -0,0 +1,141 @@
+import {forProp, handle, not, stop} from '@enact/core/handle';
+
+import {hasPointerMoved} from '../src/pointer';
+import Spotlight from '../src/spotlight';
+
+const isNewPointerPosition = (ev) => hasPointerMoved(ev.clientX, ev.clientY);
+
+class SpotlightContainer {
+	// set up context and props so we can use handle binding
+	context = {};
+	props = {};
+
+	constructor ({containerConfig, navigableFilter, preserveId}) {
+		// spotlight id - set either via props or generated by Spotlight
+		this.id = null;
+
+		// Preserve the spotlight id across instances when it is set via props and preserveId is set
+		this.canPreserveId = preserveId;
+		this.preserveId = false;
+
+		// container config - set once when the container is added
+		this.config = containerConfig;
+
+		// optional filter function to limit the spottable candidates for this container
+		this.filter = navigableFilter;
+
+		// private hash of spotlight DOM attributes
+		this.attr = {
+			'data-spotlight-container': true
+		};
+
+		// Used to indicate that we want to stop propagation on blur events that occur as a
+		// result of this component imperatively blurring itself on focus when spotlightDisabled
+		this.shouldPreventBlur = false;
+	}
+
+	get attributes () {
+		return this.attr;
+	}
+
+	setProps (props) {
+		const {disabled, id, muted, restrict} = props;
+		this.props = props;
+
+		if (this.id == null || (id && this.id !== id)) {
+			if (this.id) {
+				this.releaseContainer(this.id);
+			}
+
+			this.id = Spotlight.add({
+				...this.config,
+				id,
+				restrict,
+				navigableFilter: this.navigableFilter
+			});
+			this.preserveId = this.canPreserveId && this.id === id;
```

Comment:
Does this imply `preserveId` only works if you specify an id?  I think this behaves differently from the previous implementation insofar as a generated id would be passed to `stateFromProps`.  In this case, we should test if `id` is `null` and allow that to preserve the id.